libs/src/api/user.js

```diff
@@ -289,19 +268,24 @@ class Users extends Base {
    * @returns {Object} the passed in metadata object with profile_picture and cover_photo fields added
    */
   async uploadProfileImages (profilePictureFile, coverPhotoFile, metadata) {
+    let didMetadataUpdate = false
     if (profilePictureFile) {
       const resp = await this.creatorNode.uploadImage(profilePictureFile, true)
       metadata.profile_picture_sizes = resp.dirCID
+      didMetadataUpdate = true
     }
     if (coverPhotoFile) {
       const resp = await this.creatorNode.uploadImage(coverPhotoFile, false)
       metadata.cover_photo_sizes = resp.dirCID
+      didMetadataUpdate = true
     }
 
-    await this._handleMetadata({
-      newMetadata: metadata,
-      userId: metadata.user_id
-    })
+    if (didMetadataUpdate) {
```

Comment:
you don't need `didMetadatUpdate` since you can just check `if (profilePictureFile || coverPhotoFile)`